### PR TITLE
Add ROS2CameraSensor service and horizontal field of view param

### PR DESCRIPTION
**Pull request by adamkrawczyk**
Added:
- macro `AZ_TYPE_INFO` to the CameraDepthSensor and CameraRGBDSensor
- services to `ROS2CameraSensorComponent`
- move `CameraSensor` header to shared directory 

Solves https://github.com/o3de/o3de-extras/issues/278

### CODE REVIEW

File: Gems/ROS2/Code/Source/Camera/CameraSensor.cpp [hunk 1]
```diff
@@ -71,6 +71,8 @@ namespace ROS2
         AZ_Assert(
             m_cameraConfiguration.m_verticalFieldOfViewDeg > 0.0f && m_cameraConfiguration.m_verticalFieldOfViewDeg < 180.0f,
             "Vertical fov should be in range 0.0 < FoV < 180.0 degrees");
+        // AZ_Assert(m_cameraConfiguration.m_horizontalFieldOfViewDeg > 0.0f && m_cameraConfiguration.m_horizontalFieldOfViewDeg < 360.0f,
+        //     "Horizontal fov should be in range 0.0 < FoV < 360.0 degrees");
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/Camera/CameraSensor.cpp]: remove comment. Also, These fields should be limited in the Reflect instead, that might be a good addition within this PR to add such limits on input. Better to limit than to assert.

File: Gems/ROS2/Code/Source/Camera/CameraSensorConfiguration.h [hunk 2]
```diff
@@ -19,7 +19,8 @@ namespace ROS2
         AZ_TYPE_INFO(CameraSensorConfiguration, "{386A2640-442B-473D-BC2A-665D049D7EF5}");
         static void Reflect(AZ::ReflectContext* context);
 
-        float m_verticalFieldOfViewDeg = 90.0f; //!< Vertical field of view of camera sensor.
+        float m_horizontalFieldOfViewDeg = 86.0f; //!< Horizontal field of view of camera sensor.
+        float m_verticalFieldOfViewDeg = 57.0f; //!< Vertical field of view of camera sensor.
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/Camera/CameraSensorConfiguration.h:23]: The defaults should be referring to some representative, real, common device here. If this is the case, ignore the comment (resolve it).

File: Gems/ROS2/Code/Source/Camera/ROS2CameraSensorEditorComponent.cpp [hunk 3]
```diff
@@ -79,9 +79,9 @@ namespace ROS2
         incompatible.push_back(AZ_CRC_CE("ROS2CameraSensor"));
     }
 
-    void ROS2CameraSensorEditorComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    void ROS2CameraSensorEditorComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
     {
-        required.push_back(AZ_CRC_CE("ROS2CameraSensor"));
+        provided.push_back(AZ_CRC_CE("ROS2CameraSensor"));
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/Camera/ROS2CameraSensorEditorComponent.cpp:84]: nice catch!

File: Gems/ROS2/Code/Source/Camera/CameraSensorConfiguration.cpp [hunk 4]
```diff
@@ -32,6 +33,11 @@ namespace ROS2
                         &CameraSensorConfiguration::m_verticalFieldOfViewDeg,
                         "Vertical field of view",
                         "Camera's vertical (y axis) field of view in degrees.")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &CameraSensorConfiguration::m_horizontalFieldOfViewDeg,
+                        "Horizontal field of view",
+                        "Camera's horizontal (x axis) field of view in degrees.")
                     ->DataElement(AZ::Edit::UIHandlers::Default, &CameraSensorConfiguration::m_width, "Image width", "Image width")
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/Camera/CameraSensorConfiguration.cpp:45]: add limits here

**Review by adamdbrw — COMMENTED** [review]
Nice changes!

File: Gems/ROS2/Code/Source/Camera/CameraSensorConfiguration.h (hunk 2 above)
Comment (adamkrawczyk) [reply Gems/ROS2/Code/Source/Camera/CameraSensorConfiguration.h:23]: This are the values of the intel realsense camera (435(i))

File: Gems/ROS2/Code/Source/Camera/CameraSensor.cpp (hunk 1 above)
Comment (adamkrawczyk) [reply Gems/ROS2/Code/Source/Camera/CameraSensor.cpp]: Sorry, I haven't spotted the comment. Will remove and modify to add limits on params.

File: Gems/ROS2/Code/Source/Camera/CameraSensorConfiguration.cpp (hunk 4 above)
Comment (adamkrawczyk) [reply Gems/ROS2/Code/Source/Camera/CameraSensorConfiguration.cpp:45]: Added

**Comment by adamkrawczyk** [comment]
Closed due to opening https://github.com/o3de/o3de-extras/pull/300

**Pull request closed without merging**